@here/harp-examples/src/markers_dynamic.ts

```diff
@@ -115,78 +129,120 @@ export namespace DynamicMarkersExample {
     }
 
     let markerId = 0;
-    function attachClickEvents(mapView: MapView, markersDataSource: FeaturesDataSource) {
-        mapView.canvas.addEventListener("click", event => {
-            if (event.shiftKey) {
-                const geo = mapView.getGeoCoordinatesAt(event.clientX, event.clientY);
-                if (geo) {
-                    // Add a new marker to the data source at the click coordinates.
-                    markersDataSource.add(
-                        new MapViewPointFeature(geo.toGeoPoint() as number[], {
-                            text: markerId.toString(),
-                            id: markerId,
-                            icon: icons[markerId % icons.length].name,
-                            renderOrder: markerId
-                        })
-                    );
-                    markerId++;
-                }
-            } else if (event.ctrlKey) {
-                handlePick(mapView, markersDataSource, event.pageX, event.pageY);
-            }
-        });
+    let longTapTimer: number | undefined;
```

Comment:
Shall we move it out of the example to make it reusable and not complicate the example logic?

Reply:
done. I've adapted the LongPressHandler class to support touch events.